lib/usecases/updateApplication.js

```diff
@@ -44,33 +49,78 @@ export default async ({ clientGeneratedId, data, user }) => {
     throw error;
   }
 
-  const updateStatusQuery = `
-    INSERT INTO application_assessment(grant_application_id, application_state_id)
-    SELECT id, $1
-    FROM grant_application
-    WHERE client_generated_id = $2
-    ON CONFLICT (grant_application_id)
-    DO UPDATE
-    SET application_state_id = $1;`;
-  await dbInstance.none(updateStatusQuery, [
-    statusIndex + 1,
-    clientGeneratedId
-  ]);
+  if (data.status) {
+    return processStatus(data.status);
+  } else {
+    return processValidations(data.validations);
+  }
 
-  const addHistoryQuery = `
-    INSERT INTO application_history(grant_application_id, user_recorded, notes)
-    VALUES($(grantApplicationId), $(user), $(notes));`;
-  await dbInstance.none(addHistoryQuery, {
-    grantApplicationId,
-    user,
-    notes: `Status updated to "${data.status}"`
-  });
+  async function processValidations(validations) {
+    const updateValidationsQuery = `
+      UPDATE application_assessment
+      SET validations=$(validations)
+      WHERE grant_application_id=$(grantApplicationId) ;`;
+
+    await dbInstance.none(updateValidationsQuery, {
+      validations,
+      grantApplicationId
+    });
+
+    await addHistoryEntry(
+      dbInstance,
+      grantApplicationId,
+      user,
+      'Validations updated.'
+    );
+
+    return createResponse(null);
+  }
+
+  async function processStatus(statusString) {
+    const statusIndex = APPLICATION_STATE.findIndex(
+      status => status === statusString
+    );
+    if (statusIndex === -1) {
+      return createResponse(INVALID_STATUS);
+    }
+
+    const updateStatusQuery = `
+      INSERT INTO application_assessment(grant_application_id, application_state_id)
+      SELECT id, $1
+      FROM grant_application
+      WHERE client_generated_id = $2
+      ON CONFLICT (grant_application_id)
+      DO UPDATE
+      SET application_state_id = $1;`;
+    await dbInstance.none(updateStatusQuery, [
+      statusIndex + 1,
+      clientGeneratedId
+    ]);
```

Comment:
Shall we flip this to named parameters while we are here?

```suggestion
    const updateStatusQuery = `
      INSERT INTO application_assessment(grant_application_id, application_state_id)
      SELECT id, $(statusId)
      FROM grant_application
      WHERE client_generated_id = $(clientGeneratedId)
      ON CONFLICT (grant_application_id)
      DO UPDATE
      SET application_state_id = $(statusId);`;
    await dbInstance.none(updateStatusQuery, {
      statusId: statusIndex + 1,
      clientGeneratedId
    });
```